File: proga/1_semestr/labs/lab3/src/main/java/dev/lysmux/lab3/story/Sentence.java
package dev.lysmux.lab3.story;

public record Sentence(String content) {
    public Sentence and(String text) {
        return new Sentence(content.concat(", и ").concat(text));
    }

    public Sentence but(String text) {
        return new Sentence(content.concat(", но ").concat(text));
    }

    public Sentence that(String text) {
        return new Sentence(content.concat(", что ").concat(text));
    }

    public Sentence which(String text) {
        return new Sentence(content.concat(", которые ").concat(text));
    }

    public Sentence after(String text) {
        return new Sentence(content.concat(", после того как ").concat(text));
    }

    public Sentence comma(String text) {
        return new Sentence(content.concat(", ").concat(text));
    }
}

File: proga/1_semestr/labs/lab3/src/main/kotlin/dev/lysmux/lab3/story/Sentence.kt
package dev.lysmux.lab3.story

data class Sentence(val content: String) {
    fun and(text: String) = Sentence("$content, и $text")
    fun but(text: String) = Sentence("$content, но $text")
    fun that(text: String) = Sentence("$content, что $text")
    fun which(text: String) = Sentence("$content, которые $text")
    fun after(text: String) = Sentence("$content, после того как $text")
    fun comma(text: String) = Sentence("$content, $text")
}